CONTRIBUTING.md

```diff
@@ -0,0 +1,31 @@
+# How to successfully make changes to Ninja
+
+We're very wary of changes that increase the complexity of Ninja (in particular,
+new build file syntax or command-line flags) or increase the maintenance burden
+of Ninja. Ninja is already successfully used by hundreds of developers for large
+projects and it already achieves (most of) the goals we set out for it to do.
+It's probably best to discuss new feature ideas on the
+[mailing list](https://groups.google.com/forum/#!forum/ninja-build) or in an
+issue before creating a PR.
+
+## Coding guidelines
+
+Generally it's the
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) with
+a few additions:
+
+* Any code merged into the Ninja codebase which will be part of the main
+  executable must compile as C++03. You may use C++11 features in a test or an
+  unimportant tool if you guard your code with `#if __cplusplus >= 201103L`.
+* We have used `using namespace std;` a lot. Please try to avoid it whenever
```

Comment:
We have used `using namespace std;` a lot in the past. For new contributions, please try to avoid relying on `using namespace std;`, and instead whenever possible use `std::`. However, please do not change existing code simply to add `std::` unless your contribution already needs to change that line of code anyway. In essence, we want to avoid unnecessary code-churn, but still slowly migrate away from `using namespace std;`

Reply:
IMHO that's too long and the original message brings the point across already.

Reply:
I don't think that the original message brings the point across effectively.

As a native English speaker, I did not feel that the original statement was easy to understand, and needed to rely on other discussions that you and I have had to understand what was being communicated.

Reply:
Okay, I'll come up with a modified version then, based on your one.